Content.Client/VendingMachines/VendingMachineBoundUserInterface.cs

```diff
@@ -26,26 +27,79 @@ protected override void Open()
             _menu = this.CreateWindowCenteredLeft<VendingMachineMenu>();
             _menu.Title = EntMan.GetComponent<MetaDataComponent>(Owner).EntityName;
             _menu.OnItemSelected += OnItemSelected;
-            Refresh();
+            
+            Refresh(); // 🌟Starlight🌟
+            
+            // 🌟Starlight🌟 
+            if (_showPrices)
+            {
+                RequestBalance(); // Client ask too, server also pushes on open now
+            }
+        }
+
+        protected override void ReceiveMessage(BoundUserInterfaceMessage message)
+        {
+            if (message is VendingMachineBalanceUpdateMessage balanceMessage)
+            {
+                _menu?.UpdateBalance(balanceMessage.Balance);
+            }
+        }
+
+        /// <summary>
+        /// Requests current balance from server
+        /// </summary>
+        private void RequestBalance()         // 🌟Starlight🌟
+        {
+            SendMessage(new VendingMachineRequestBalanceMessage());
         }
 
         public void Refresh()
         {
             var enabled = EntMan.TryGetComponent(Owner, out VendingMachineComponent? bendy) && !bendy.Ejecting;
+            _showPrices = bendy?.ShowPrices ?? true;
 
             var system = EntMan.System<VendingMachineSystem>();
             _cachedInventory = system.GetAllInventory(Owner);
 
-            _menu?.Populate(_cachedInventory, enabled);
+            _menu?.Populate(_cachedInventory, enabled, _showPrices);
+            
+            // 🌟Starlight🌟 
+            if (_menu != null)
+            {
+                if (_showPrices)
+                {
+                    _menu.ShowBalance();
+                    RequestBalance();  // 🌟Starlight🌟
```

Comment:
```suggestion
                    RequestBalance();
```

Starlight in starlight?